### PR TITLE
[luci/pass] Trim RemoveUnnecessaryTransposesNetPass

### PR DESCRIPTION
**Pull request by zetwhite**
This PR trims RemoveUnnecessaryTransposesNetPass.
- enhance some comements
- rename a varaible
- replace 'if(not cond) return false' with macro

ONE-DCO-1.0-Signed-off-by: seunghui youn <sseung.youn@samsung.com>

### CODE REVIEW

File: compiler/luci/pass/src/RemoveUnnecessaryTransposeNetPass.cpp [hunk 1]
```diff
@@ -233,16 +236,16 @@ bool TaggedShapeAnalyzer::analyze_reshape(const std::vector<int32_t> &new_shape)
...
+    auto target_dim = new_shape[target_idx];
 
     // Ignore dim == 1
     if (target_dim == 1)
     {
       new_tagged_shape.emplace_back(1);
-      new_shape_idx++;
+      target_idx++;
       continue;
     }
```
Comment (zetwhite) [inline compiler/luci/pass/src/RemoveUnnecessaryTransposeNetPass.cpp:250]: rename `new_shape_idx` -> `target_idx

File: compiler/luci/pass/src/RemoveUnnecessaryTransposeNetPass.cpp [hunk 2]
```diff
@@ -467,24 +470,16 @@ bool remove_unnecessary_transpose(luci::CircleTranspose *node)
...
 
   TaggedShapeAnalyzer analyzer;
-
-  if (not analyzer.init(front_transpose, mid_reshape, back_transpose))
-    return false;
-
-  if (not analyzer.can_remove_transposes())
-    return false;
+  CHECK_OR_FALSE(analyzer.init(front_transpose, mid_reshape, back_transpose));
+  CHECK_OR_FALSE(analyzer.can_remove_transposes());
```
Comment (zetwhite) [inline compiler/luci/pass/src/RemoveUnnecessaryTransposeNetPass.cpp:482]: (note) 

replace `if (not cond) return false` with `CHECK_OR_FALSE` macro.

File: compiler/luci/pass/src/RemoveUnnecessaryTransposeNetPass.cpp [hunk 3]
```diff
@@ -329,7 +332,7 @@ bool TaggedShapeAnalyzer::init(const luci::CircleTranspose *front_transpose,
     return true;
   };
 
-  // check c3
+  // check c2
```
Comment (zetwhite) [inline compiler/luci/pass/src/RemoveUnnecessaryTransposeNetPass.cpp:335]: (note) re-numbering condition
`c1` is removed in previous PR - https://github.com/Samsung/ONE/pull/14167#discussion_r1789782341

**Review by jinevening — APPROVED** [review]
LGTM

**Pull request merged**